lib/global-admin/addon/components/modal-edit-catalog/component.js

```diff
@@ -25,7 +33,23 @@ export default Component.extend(ModalBase, NewOrEdit, {
     set(clone, 'kind', 'helm');
     set(this, 'model', clone);
   },
+  validate: function() {
+    const intl = get(this, 'intl');
+    var model = get(this, 'primaryResource');
+    var name = get(this, 'primaryResource.name')
+    var errors = model.validationErrors();
+    if (name && existingCatalogs.indexOf(name.toLowerCase()) > -1) {
```

Comment:
`existingCatalogs.includes(name.toLowerCase())`.  But this logic is wrong in general... you can't hardcode existingCatalogs, it should check against the ones that actually exist right now.

If the standard `library` is disabled then you should to be able to add your own catalog with the name library and an arbitrary URL.

And if I already have a catalog named 'foo', you might as well not let me add a second one named that since the API is going to say no anyway.